### PR TITLE
Refactored white_space.py

### PR DESCRIPTION
**Pull request by rahulroxx**

### CODE REVIEW

File: white_space.py [hunk 1]
```diff
@@ -22,7 +21,7 @@ def format_file(path):
     '''
     p = ""
     for dirpath, _, filenames in os.walk(path):
-        if(os.path.exists(os.path.join(dirpath, 'bin/activate'))):
+        if (os.path.exists(os.path.join(dirpath, 'bin/activate'))):
```
Comment (goelakash) [inline white_space.py]: Make this as:
`if os.path.exists(os.path.join(dirpath, 'bin/activate')):`

**Comment by rahulroxx** [comment]
@goelakash Thanks! Please review and test it .

**Comment by goelakash** [comment]
@rahulroxx Please squash these commits.

File: white_space.py [hunk 2]
```diff
@@ -22,24 +21,23 @@ def format_file(path):
...
-                            source_code = file1.read()
-                            file1.seek(0)
-                            file1 = open(filename, 'w')
-                            file1.truncate()
-                            file1.write(autopep8.fix_code(source_code))
-                            file1.close()
-                        except:
-                            print(filename)
-                            print(sys.exc_info()[0])
+        for file in filenames:
```
Comment (CuriousLearner) [inline white_space.py]: @rahulroxx Please don't use `file` as a variable name. It's a built-in in python.

Till the time we're supporting python 2.x it's not a good thing to use file as a variable name.

Comment (goelakash) [inline white_space.py]: Completely missed that :worried:

**Comment by rahulroxx** [comment]
Oops ! will correct that Thanks @CuriousLearner .

**Comment by rahulroxx** [comment]
Apologies, Please correct me if i am wrong but I couldn't find file as builtin data type in python 2.x, 

Try This, python 2.7.x 
```
>>> import keyword
>>> keyword.iskeyword('file')
False
```
Can you please mention the release.

**Comment by aktech** [comment]
It's not a keyword, it's a built-in.
https://docs.python.org/2/library/functions.html

**Comment by rahulroxx** [comment]
@aktech Thanks!

**Comment by rahulroxx** [comment]
@aktech Have a look now changed 

> file

 to

>  file_

File: white_space.py [hunk 3]
```diff
@@ -22,24 +21,23 @@ def format_file(path):
...
-                            file1 = open(filename, 'w')
-                            file1.truncate()
-                            file1.write(autopep8.fix_code(source_code))
-                            file1.close()
-                        except:
-                            print(filename)
-                            print(sys.exc_info()[0])
+        for file_ in filenames:
+            if file_.endswith('.py'):
+                file_ = os.path.abspath(os.path.join(dirpath, file_))
```
Comment (goelakash) [inline white_space.py]: You are reassigning loop variable here, which is bad programming practice.
I suggest using `f` or some other keyword for looping variable, and keep `file_` here as is.

**Comment by rahulroxx** [comment]
Thanks @goelakash !

**Pull request closed without merging**